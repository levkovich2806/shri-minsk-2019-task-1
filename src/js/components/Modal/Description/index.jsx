import React from "react";
import styles from "./index.module.scss";

function Description(props) {
  return (
    <div className={styles.description}>
      <span className={styles.description__title}>Описание: </span>
      <textarea
        className={styles.description__area}
        onChange={props.onChange}
      />
    </div>
  );
}

export default Description;
