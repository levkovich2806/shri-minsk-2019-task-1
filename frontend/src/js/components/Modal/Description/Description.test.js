import React from 'react';
import { shallow } from 'enzyme';

import Description from './index';

describe('Description', () => {
  it('Should match snapshot description', () => {
    const wrapper = shallow(<Description description="Описание" />);
    expect(wrapper).toMatchSnapshot();
  });
});
