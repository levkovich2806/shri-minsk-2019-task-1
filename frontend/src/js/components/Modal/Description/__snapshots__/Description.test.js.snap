// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description Should match snapshot description 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Description
    description="Описание"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="description__title"
        >
          Описание: 
        </span>,
        <textarea
          className="description__area"
          value="Описание"
        />,
      ],
      "className": "description",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Описание: ",
          "className": "description__title",
        },
        "ref": null,
        "rendered": "Описание: ",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "description__area",
          "onChange": undefined,
          "value": "Описание",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="description__title"
          >
            Описание: 
          </span>,
          <textarea
            className="description__area"
            value="Описание"
          />,
        ],
        "className": "description",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Описание: ",
            "className": "description__title",
          },
          "ref": null,
          "rendered": "Описание: ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "description__area",
            "onChange": undefined,
            "value": "Описание",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
